test/opengl/produce.cpp

```diff
@@ -0,0 +1,120 @@
+#include "Halide.h"
+#include <stdio.h>
+#include <stdlib.h>
+
+using namespace Halide;
+
+// This test creates two input images and uses one to perform a dependent lookup
+// into the other. The lookup table is produced using a Halide func scheduled
+// on the host.
+
+int test_lut1d() {
+
+    Var x("x");
+    Var y("y");
+    Var c("c");
+
+    Image<uint8_t> input(8, 8, 3, "input");
+    for (int y = 0; y < input.height(); y++) {
+        for (int x = 0; x < input.width(); x++) {
+            float v = (1.0f / 16.0f) + (float)x / 8.0f;
+            input(x, y, 0) = (uint8_t)(v * 255.0f);
+            input(x, y, 1) = (uint8_t)((1.0f - v) * 255.0f);
+            input(x, y, 2) = (uint8_t)((v > 0.5 ? 1.0 : 0.0) * 255.0f);
+        }
+    }
+
+    // 1D Look Up Table case
+    Func lut1d("lut1d");
+    lut1d(x) = cast<float>(1 + x);
+
+    Func f0("f");
+    Expr e = cast<int>(8.0f * cast<float>(input(x, y, c)) / 255.0f);
+
+    f0(x, y, c) = lut1d(clamp(e, 0, 7));
+    lut1d.compute_root();
+
+    f0.bound(c, 0, 3);
+    f0.glsl(x, y, c);
+
+    Image<float> out0(8, 8, 3, "out");
+    f0.realize(out0);
+
+    out0.copy_to_host();
+
+#if 0
```

Comment:
We shouldn't generally check in commented-out code like this... if there's a good reason (example of what not to do, etc) then add a comment to that effect. If not, better to save it elsewhere.